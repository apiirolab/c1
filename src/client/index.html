<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>