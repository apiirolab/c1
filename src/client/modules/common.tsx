import * as React from 'react';
import { QueryResult } from 'react-apollo';
import { List, Loader, Message } from 'semantic-ui-react';
import { context } from '../config/context';

export function renderResult(
  result: QueryResult<any, any>,
  Component: () => JSX.Element
): JSX.Element {
  if (result.loading) {
    return (
      <div>
        <Loader inline size="tiny" active={true} />
        &nbsp;
        {context.i18n` Loading ...`}
      </div>
    );
  }
  if (result.error) {
    return (
      <p>
        {result.error.graphQLErrors &&
          result.error.graphQLErrors.length && (
            <Message
              error={true}
              header="GraphQL Errors"
              content={
                <List>
                  {result.error.graphQLErrors.map((e, i) => (
                    <List.Item key={i}>{e.message}</List.Item>
                  ))}
                </List>
              }
            />
          )}
        {result.error.networkError && (
          <Message
            error={true}
            header="Network Error"
            content={JSON.stringify(result.error.networkError)}
          />
        )}
      </p>
    );
  }
  return Component();
}
