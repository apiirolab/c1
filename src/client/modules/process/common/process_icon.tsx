import * as React from 'react';

import { List, SemanticICONS } from 'semantic-ui-react';
import styled from 'styled-components';

export function processTypeToIcon(type: string): SemanticICONS {
  switch (type) {
    case 'Travel':
      return 'briefcase';
    case 'HumanResources':
      return 'user circle';
    case 'Purchases':
      return 'money';
    case 'Sales':
      return 'money bill alternate outline';
  }
  return 'tasks';
}

const IconView = styled(List.Icon)`
  &&&&& {
    vertical-align: top !important;
    width: 25px;
  }
`;

export type Props = {
  type: string;
};

export const ProcessIcon: React.SFC<Props> = ({ type }) => (
  <IconView name={processTypeToIcon(type)} verticalAlign="middle" />
);
