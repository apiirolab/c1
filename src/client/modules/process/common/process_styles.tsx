import { Grid } from 'semantic-ui-react';
import styled from 'styled-components';

export type EditableViewType = 'view' | 'preview';

export type ProcessViewType =
  | 'information'
  | 'resources'
  | 'form'
  | 'report'
  | 'document'
  | 'diagram'
  | 'log';

export const TightGrid = styled(Grid)`
  margin: 12px 0px 0px 0px !important;
  .column {
    padding-top: 5px !important;
  }
  .label {
    margin-left: 0px !important;
  }
`;

export const TabContent = styled.div`
  &&&&& {
    padding: 12px;
    height: 100%;

    .header .content {
      width: 100%;
    }
    /* background-image: linear-gradient(#d4d4d5, #ffffff), linear-gradient(#d4d4d5, #ffffff);
    background-size: 1px 100%;
    background-position: 0 0, 100% 0;
    background-repeat: no-repeat; */
  }
`;
