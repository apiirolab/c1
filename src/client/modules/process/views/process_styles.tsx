import styled from 'styled-components';

export const TabContent = styled.div`
  &&&&& {
    padding: 12px;
    height: 100%;
    /* background-image: linear-gradient(#d4d4d5, #ffffff), linear-gradient(#d4d4d5, #ffffff);
    background-size: 1px 100%;
    background-position: 0 0, 100% 0;
    background-repeat: no-repeat; */
  }
`;
