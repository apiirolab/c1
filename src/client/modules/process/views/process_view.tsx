import * as React from 'react';
import {
  Header,
  Icon,
  Grid,
  Segment,
  List,
  Button,
  ListProps,
  TextArea,
  TextAreaProps
} from 'semantic-ui-react';

import * as Diagram from './PatentApplicationHighLevel.png';
import styled, { StyledComponentClass } from 'styled-components';

const LogList: StyledComponentClass<ListProps, {}> = styled(List)`
  &&&&& .header {
    font-weight: normal;
  }
  &&&&& .description {
    font-size: smaller;
  }
`;

const CommentArea: StyledComponentClass<TextAreaProps, {}> = styled(TextArea)`
  &&&&& {
    margin-bottom: 12px;
  }
`;

interface Props {
  running: boolean;
}

const Roles: React.SFC<Props> = ({ running }) => (
  <>
    <Header inverted attached as="h5" icon="users" content="Roles" dividing />
    <Segment attached="bottom">
      <List>
        <List.Item as="a" icon="user" content="Dean &middot; Simi Simivov" />
        <List.Item as="a" icon="user" content="Travel Team &middot; Andrew Bold" />
      </List>
    </Segment>
  </>
);

const Log = () => (
  <>
    <Header inverted as="h5" icon="tasks" content="Log" dividing attached />
    <Segment attached>
      <LogList divided relaxed>
        <List.Item>
          <List.Icon name="hourglass half" verticalAlign="middle" />
          <List.Content>
            <List.Header>Awaiting dean's approval</List.Header>
          </List.Content>
        </List.Item>
        <List.Item>
          <List.Icon name="check" verticalAlign="middle" />
          <List.Content>
            <List.Header>I have found no issues in the report. Proposing acceptance.</List.Header>
            <List.Description as="a">
              Approved by Angie Law &middot; May 27 at 11:22am
            </List.Description>
          </List.Content>
        </List.Item>
        <List.Item>
          <List.Icon name="ticket" verticalAlign="middle" />
          <List.Content>
            <List.Description as="a">
              Submitted by Tomas Trescak &middot; May 25 at 10:15am
            </List.Description>
          </List.Content>
        </List.Item>
      </LogList>
    </Segment>
  </>
);

const Resources = () => (
  <>
    <Header inverted attached="top" as="h5" icon="file" content="Resources" dividing />
    <Segment attached>
      <List>
        <List.Item>
          <List.Icon name="folder" />
          <List.Content>
            <List.Header>Forms</List.Header>
            <List.Description>User forms of this process</List.Description>
            <List.List>
              <List.Item icon="archive" content="Travel Request" />
              <List.Item icon="archive" content="DTS" />
            </List.List>
          </List.Content>
        </List.Item>
        <List.Item>
          <List.Icon name="folder" />
          <List.Content>
            <List.Header>Reports and Receipts</List.Header>
            <List.List>
              <List.Item icon="newspaper" content="Dean's Report" />
            </List.List>
          </List.Content>
        </List.Item>
        <List.Item>
          <List.Icon name="folder" />
          <List.Content>
            <List.Header>Resources</List.Header>
            <List.List>
              <List.Item icon="file word outline" content="TEMS Instructions" />
              <List.Item icon="file excel outline" content="Max funding" />
            </List.List>
          </List.Content>
        </List.Item>
      </List>
    </Segment>
  </>
);

export const ProcessView: React.SFC<Props> = ({ running }) => (
  <>
    <Grid className="ui form">
      <Grid.Column width={11}>
        <Header as="h2">
          <Icon name="travel" />
          <Header.Content>
            Domestic Travel
            {/* <Header.Subheader>Manage your preferences</Header.Subheader> */}
          </Header.Content>
        </Header>

        <p>
          This process allows you to book a short domestic travel with length up to 3 working days.
          The purpose of travel is either business or academic. The cost centre needs to be known in
          advance to proceed with booking. Please provide also the detailed travel schedule.
        </p>

        <If condition={running}>
          <>
            <Segment inverted color="green">
              <Header content="Running: Awaiting Input" as="h5" icon="tasks" dividing />
              <List inverted>
                <List.Item as="a">
                  <Icon name="help" />
                  <List.Content>
                    <List.Header>Awaiting your decision</List.Header>
                    <List.Description>
                      Please approve or reject this travel request
                    </List.Description>
                  </List.Content>
                </List.Item>
                <List.Item as="a">
                  <Icon name="check" />
                  Approved by Ange Arthur &middot; 3 hours ago &middot; All details are correct
                </List.Item>
              </List>
            </Segment>

            <CommentArea rows={2} placeholder="Please leave comment ..." />

            <Button color="red" icon="remove" content="Reject" floated="right" />
            <Button color="green" icon="check" content="Accept" floated="right" />
            <div style={{ clear: 'both' }} />
          </>
        </If>

        <Header icon="sitemap" content="Process Diagram" as="h4" />
        <img src={Diagram} style={{ width: '100%' }} />
      </Grid.Column>
      <Grid.Column width={5}>
        <Resources />
        <If condition={running}>
          <Log />
        </If>
        <Roles running={running} />
      </Grid.Column>
    </Grid>
  </>
);
