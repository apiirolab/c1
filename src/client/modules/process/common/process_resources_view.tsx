import * as React from 'react';

import { List } from 'semantic-ui-react';

import { Link } from '@reach/router';
import { QueryTypes } from 'data/client';

type Props = {
  context: App.Context;
  process: QueryTypes.BpmnProcessDefinition;
  processInstance: QueryTypes.BpmnProcessInstance;
  filter?: QueryTypes.ResourceType;
};

export class ProcessResources extends React.Component<Props> {
  // renderModal() {
  //   const resource = this.state.activeResource;

  //   return (
  //     <Modal
  //       open={true}
  //       dimmer="blurring"
  //       size="fullscreen"
  //       closeOnDimmerClick={false}
  //       closeOnEscape={false}
  //     >
  //       <Modal.Header>{resource.name}</Modal.Header>
  //       <Modal.Content scrolling>
  //         description
  //         <br />
  //       </Modal.Content>
  //       <Modal.Actions>
  //         <Button
  //           default
  //           content={this.props.context.i18n`Close`}
  //           onClick={() => this.setState({ activeResource: null })}
  //         />
  //       </Modal.Actions>
  //     </Modal>
  //   );
  // }

  createUrl(r: QueryTypes.BpmnProcessResource, id: String) {
    return this.props.processInstance
      ? `/process/${this.props.process.name.url()}/view/${r.type.toLowerCase()}/${r.name.url()}/${
          this.props.processInstance.id
        }/${id}`
      : `/process/${this.props.process.name.url()}/preview/${r.type.toLowerCase()}/${r.name.url()}/${
          this.props.process.id
        }/${id}`;
  }

  render() {
    const { process, filter } = this.props;

    const forms = process.resources.filter(r => r.type === QueryTypes.ResourceType.Form);
    const reports = process.resources.filter(r => r.type === QueryTypes.ResourceType.Report);
    const documents = process.resources.filter(r => r.type === QueryTypes.ResourceType.Document);
    const files = process.resources.filter(r => r.type === QueryTypes.ResourceType.File);
    const links = process.resources.filter(r => r.type === QueryTypes.ResourceType.Url);

    return (
      <>
        {/* {this.state.activeResource && this.renderModal()} */}

        <If condition={!process.resources.length}>
          <I18>ðŸ“‰ No resources allocated to this process</I18>
        </If>
        <List>
          <If condition={!!forms.length && (!filter || filter === QueryTypes.ResourceType.Form)}>
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Forms</List.Header>
                {/* <List.Description>User forms of this process</List.Description> */}

                <List.List>
                  {forms.map(r => (
                    <List.Item
                      as={Link}
                      to={this.createUrl(r, (r.form || { id: 'error' }).id)}
                      key={r.id}
                      icon="wordpress forms"
                      content={r.name}
                      // onClick={() => this.setState({ activeResource: r })}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
          <If
            condition={!!reports.length && (!filter || filter === QueryTypes.ResourceType.Report)}
          >
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Reports</List.Header>
                <List.List>
                  {reports.map(r => (
                    <List.Item
                      as={Link}
                      to={this.createUrl(r, (r.form || { id: 'error' }).id)}
                      key={r.id}
                      icon="newspaper"
                      content={r.name}
                      onClick={() => this.setState({ activeResource: r })}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
          <If
            condition={
              !!documents.length && (!filter || filter === QueryTypes.ResourceType.Document)
            }
          >
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Documents</List.Header>
                <List.List>
                  {documents.map(r => (
                    <List.Item
                      as={Link}
                      to={this.createUrl(r, r.document.id)}
                      key={r.id}
                      icon="file"
                      content={r.name}
                      onClick={() => this.setState({ activeResource: r })}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
          <If condition={!!files.length && !filter}>
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Files</List.Header>
                <List.List>
                  {files.map(r => (
                    <List.Item
                      as={Link}
                      target="_blank"
                      key={r.id}
                      icon="file alternate"
                      content={r.name}
                      to={`/files/${process.id}/${r.link}`}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
          <If condition={!!links.length && !filter}>
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Links</List.Header>
                <List.List>
                  {links.map(r => (
                    <List.Item
                      as="a"
                      target="_blank"
                      key={r.id}
                      icon="linkify"
                      content={r.name}
                      href={r.link}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
        </List>
      </>
    );
  }
}
