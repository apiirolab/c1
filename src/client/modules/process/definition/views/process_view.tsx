import * as React from 'react';
import { Grid } from 'semantic-ui-react';

import { QueryTypes } from 'data/client';
import { ProcessViewType, TightGrid } from '../../common/process_styles';
import { MenuResourceView } from '../../menu/menu_resources_view';
import { TabDiagramView } from '../../tabs/views/tab_diagram_view';
import { TabProcessDescription } from '../../tabs/views/tab_process_description';
import { TabProcessResources } from '../../tabs/views/tab_process_resources';

interface Props {
  process: QueryTypes.BpmnProcessDefinition;
  context: App.Context;
  view: ProcessViewType;
  resourceId?: string;
}

export class ProcessView extends React.Component<Props> {
  static displayName = 'ProcessView';

  render() {
    const { process, context, view } = this.props;
    const isResource = ['resources', 'form', 'report', 'document'].includes(view);

    return (
      <div className="pageFrom">
        <Choose>
          <When condition={view === 'information'}>
            <TightGrid className="ui form" stackable>
              <Grid.Column width={view === 'information' && process.resources.length ? 11 : 16}>
                <TabProcessDescription process={process} context={context} />
              </Grid.Column>

              <If condition={!!process.resources.length}>
                <Grid.Column width={5}>
                  <MenuResourceView context={context} process={process} />
                </Grid.Column>
              </If>
            </TightGrid>
          </When>

          <When condition={isResource}>
            <TabProcessResources
              process={process}
              context={context}
              contentType={view}
              resourceId={this.props.resourceId}
              viewType={'preview'}
              ownerId={process.id}
            />
          </When>

          <When condition={view === 'diagram'}>
            <TabDiagramView
              context={context}
              process={process}
              ownerId={process.id}
              viewType="preview"
            />
          </When>
        </Choose>
      </div>
    );
  }
}
