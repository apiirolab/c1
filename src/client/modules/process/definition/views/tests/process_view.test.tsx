import * as React from 'react';

import 'jest-styled-components';

import { ProcessViewType } from 'client/modules/process/common/process_styles';
import { createProcess } from 'client/modules/process/common/tests/process_test_data';
import { create, QueryTypes, render } from 'client/tests';
import { ProcessView } from '../process_view';

describe('Process Tabs', () => {
  const context = create.context();

  function componentWithData(
    process: QueryTypes.BpmnProcessDefinition,
    view: ProcessViewType = 'information'
  ) {
    return <ProcessView context={context} process={process} view={view} />;
  }

  describe('Information', () => {
    describe('No Resources', () => {
      const component = componentWithData(createProcess({ resources: [] }));

      it('renders', () => {
        const rendered = render(component);
        expect(rendered).toMatchSnapshot();
      });

      return { component };
    });

    describe('With Resources', () => {
      const component = componentWithData(createProcess());

      it('renders', () => {
        const rendered = render(component);
        expect(rendered).toMatchSnapshot();
      });

      return { component };
    });
  });
});
