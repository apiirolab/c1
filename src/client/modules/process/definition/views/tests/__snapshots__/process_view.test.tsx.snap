// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process Tabs Information No Resources renders 1`] = `
<style>
.c0 {
  margin: 12px 0px 0px 0px !important;
}

.c0 .column {
  padding-top: 5px !important;
}

.c0 .label {
  margin-left: 0px !important;
}

.c1.c1.c1.c1.c1 {
  padding: 12px;
  height: 100%;
}

.c1.c1.c1.c1.c1 .header .content {
  width: 100%;
}</style>

<div
  className="pageFrom"
>
  <div
    className="ui stackable grid ui form c0"
  >
    <div
      className="sixteen wide column"
    >
      <div
        className="c1"
      >
        <h2
          className="ui header"
        >
          <i
            aria-hidden="true"
            className="briefcase icon"
          />
          <div
            className="content"
          >
            Process
            <a
              className="ui mini icon right floated left labeled button"
              href="/process/process/preview/diagram/1"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="sitemap icon"
              />
              Show Diagram
            </a>
            <div
              className="sub header"
            >
              Manage your process · 
              Travel
            </div>
          </div>
        </h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
        <button
          className="ui primary button"
          onClick={[Function]}
          role="button"
        >
          <i
            aria-hidden="true"
            className="rocket icon"
          />
          Launch Process
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Process Tabs Information With Resources renders 1`] = `
<style>
.c0 {
  margin: 12px 0px 0px 0px !important;
}

.c0 .column {
  padding-top: 5px !important;
}

.c0 .label {
  margin-left: 0px !important;
}

.c1.c1.c1.c1.c1 {
  padding: 12px;
  height: 100%;
}

.c1.c1.c1.c1.c1 .header .content {
  width: 100%;
}</style>

<div
  className="pageFrom"
>
  <div
    className="ui stackable grid ui form c0"
  >
    <div
      className="eleven wide column"
    >
      <div
        className="c1"
      >
        <h2
          className="ui header"
        >
          <i
            aria-hidden="true"
            className="briefcase icon"
          />
          <div
            className="content"
          >
            Process
            <a
              className="ui mini icon right floated left labeled button"
              href="/process/process/preview/diagram/1"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="sitemap icon"
              />
              Show Diagram
            </a>
            <div
              className="sub header"
            >
              Manage your process · 
              Travel
            </div>
          </div>
        </h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
        <button
          className="ui primary button"
          onClick={[Function]}
          role="button"
        >
          <i
            aria-hidden="true"
            className="rocket icon"
          />
          Launch Process
        </button>
      </div>
    </div>
    <div
      className="five wide column"
    >
      <h5
        className="ui tiny dividing inverted top attached header"
      >
        <i
          aria-hidden="true"
          className="folder icon"
        />
        <div
          className="content"
        >
          Resources
        </div>
      </h5>
      <div
        className="ui attached segment"
      >
        <div
          className="ui list"
          role="list"
        >
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <i
              aria-hidden="true"
              className="folder icon"
            />
            <div
              className="content"
            >
              Forms
              <div
                className="list"
              >
                <a
                  className="item"
                  href="/process/process/preview/form/approval-form/1/1"
                  onClick={[Function]}
                  role="listitem"
                >
                  <i
                    aria-hidden="true"
                    className="wordpress forms icon"
                  />
                  <div
                    className="content"
                  >
                    Approval Form
                  </div>
                </a>
                <a
                  className="item"
                  href="/process/process/preview/form/approval-report/1/1"
                  onClick={[Function]}
                  role="listitem"
                >
                  <i
                    aria-hidden="true"
                    className="wordpress forms icon"
                  />
                  <div
                    className="content"
                  >
                    Approval Report
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <i
              aria-hidden="true"
              className="folder icon"
            />
            <div
              className="content"
            >
              Documents
              <div
                className="list"
              >
                <a
                  className="item"
                  href="/process/process/preview/document/guidelines/1/1"
                  onClick={[Function]}
                  role="listitem"
                >
                  <i
                    aria-hidden="true"
                    className="file icon"
                  />
                  <div
                    className="content"
                  >
                    Guidelines
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <i
              aria-hidden="true"
              className="folder icon"
            />
            <div
              className="content"
            >
              Files
              <div
                className="list"
              >
                <a
                  className="item"
                  href="/files/1//uploads/file.doc"
                  onClick={[Function]}
                  role="listitem"
                  target="_blank"
                >
                  <i
                    aria-hidden="true"
                    className="file alternate icon"
                  />
                  <div
                    className="content"
                  >
                    Excel File
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <i
              aria-hidden="true"
              className="folder icon"
            />
            <div
              className="content"
            >
              Links
              <div
                className="list"
              >
                <a
                  className="item"
                  href="http://google.com"
                  onClick={[Function]}
                  role="listitem"
                  target="_blank"
                >
                  <i
                    aria-hidden="true"
                    className="linkify icon"
                  />
                  <div
                    className="content"
                  >
                    External Resource
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
